File: codeforces/c_71a_Way_Too_Long_Words.cc
#include <iostream>
#include <string>

int main()
{
    using namespace std;
    int n;
    cin >> n;
    while(n--)
    {
        string s;
        cin >> s;
        if (s.size() > 10)
            s =  s[0] + to_string(s.size()-2) +s[s.size()-1];
        cout << s << endl;
    }
}

File: codeforces/c_71a_Way_Too_Long_Words.rs
use std::io;

fn main(){
    let mut s:String = String::new();
    io::stdin().read_line(&mut s).unwrap();
    let n = s.trim().len();
    if n >= 10
    {
        println!("{}", s[0]+n.to_string()+s.chars().last());
    }
}